### PR TITLE
Complete observable immediately after emission

### PR DESCRIPTION
**Pull request by fvoska**
This PR completes the observable after emission, as is expected when making API calls in Angular.

I also removed the internal `takeUntil$` property from the response snapshot. It caused the tests to fail and we probably don't want to expose that anyway.

I had to bump TS target from es5 to es6 because of some limitation with accessing super's properties when targeting es5 (more detailed explanation below). If we want to keep es5 target for browser compatibility reasons, we have to copy `snapshot` getter and serializeHeaders helper from the parent class and execute the same `delete` command at the end. We can discuss which of those 2 options we want.

----

This will work correctly if your target is es6 or above:
```ts
class Foo {
    private secret = 'foo'

    public get notSoSecret(): string {
        return this.secret.toUpperCase();
    }
}

class Bar extends Foo {
    public get notSoSecret(): string {
        const originalSecret = super.notSoSecret;

        return `! ${originalSecret} !`
    }
}

const b = new Bar();
console.log(b.notSoSecret); // "! FOO !"
```

However, if your target is es5, you can not access `super.notSoSecret`, it says that `Only public and protected methods of the base class are accessible via the 'super' keyword.`

In other words, you can not access any properties of the super class (nor ordinary properties nor getters), you can only call super's methods.

This is because apparently there was no good way to transpile this behaviour down to ES5 (https://github.com/microsoft/TypeScript/issues/338#issuecomment-54731018)

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQA4CcCWAbmAC4Cm0EpwOpx0AvNAOQBmiTq6mWArgEYg8waAHNa0AHbxiAZXgyqNYgAoAlAC4KxfBJEoumTEp44J0YgAs8EAHSVqtG8XgBVLFlI4AwpFJqA3AbQAL6ooaigkDAAQmA40KQAHmQSACYwCEhohrwCQqLiUrLyirRqmhDaeLr6hobA8BKV0PD4ItVgIAoOdIwQPB44NkVy3UqBQYbGptAABgCE0AAkyK147RKdY7TB0POzQaHhDU10fAySpADu0LE4ARGNEPAgpDYg8CLKfMPSo6XEVT+IA

### CODE REVIEW

**Review by kristijanbambir — APPROVED** [review]

**Review by safo6m — APPROVED** [review]

File: packages/datx-jsonapi-angular/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@datx/jsonapi-angular",
-  "version": "2.1.1",
+  "version": "2.1.2",
```
Comment (DarkoKukovec) [inline packages/datx-jsonapi-angular/package.json:3]: You shouldn't update this. It's done automatically by lerna during the publish process

**Review by DarkoKukovec — CHANGES_REQUESTED** [review]

**Review by DarkoKukovec — APPROVED** [review]
